### PR TITLE
Incomplete week-5 assessment. Need to do JS challenges and Jest tests.

### PR DESCRIPTION
**Pull request by jayaries329**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Comment by jayaries329** [comment]
JS challenges and Jest tests have been submitted

File: code-challenges.test.js [hunk 1]
```diff
@@ -21,11 +21,75 @@ const secretCodeWord2 = "Gobbledygook"
...
+// Create a function named messageCoder that takes in a string
+// Use a combination of built-in methods .map, .split, .join, and a conditional statement for iteration and conversions
+// Return a coded message
+
+const messageCoder = (string) => {
+    arrayMorph = string.split("")
+    const vowelConvert = arrayMorph.map(value => {
+        if (value === 'a' || value === 'A') {
+            return '4';
+        } if (value === 'e' || value === 'E') {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:66]: good work. Best practices is to have one `if` statement, one `else` statement, and as many `else if` statements as necessary. The `else if` statements are written in the middle. We begin with the `if`. It is optional to have an `else`, but when you do, the `else` will be the last statement.

File: code-challenges.test.js [hunk 2]
```diff
@@ -34,10 +98,58 @@ const letterA = "a"
...
+//     107 |         
+
+//     at Object.expect (code-challenges.test.js:104:9)
+
+
 // b) Create the function that makes the test pass.
+// Pseudo code:
+// Create a function named fruitByLetter that takes in an array and a string
+// Use a combination of built-in methods .filter, .includes, and .toLowerCase in a high-order function for iteration and element selection based on letter
+// Return edited, shorter array
```
Comment (SunkissedQueen) [inline code-challenges.test.js:134]: Remember input, expected output, and what you are accessing to move from the input to the output

File: interview-questions.md [hunk 3]
```diff
@@ -9,43 +9,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a hash in Ruby?
 
 Your answer:
+A hash in Ruby is like an array in JavaScript because it stores data in key-value pairs, but hashes don't use numbers as an index.
 
 Researched answer:
+Hashes can also be look at as dictionaries because they perform like them. The term iterate and its definition are an example of a hash:
```
Comment (SunkissedQueen) [inline interview-questions.md:15]: Cool. More content can be sharing ruby hash to support this dictionary connection

File: interview-questions.md [hunk 4]
```diff
@@ -9,43 +9,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+iterate - To execute a set of actions repeatedly to accomplish a task.
+In this example, the term "iterate" is the key and the definition that follows is the value. We access the data by looking up the word in the dictionary
 
 2. What is a gem?
 
 Your answer:
+A gem is Ruby code that has been made available for developers to use. Ruby on Rails is an example of a Ruby gem. Rails is a package of Ruby code that is made available to use in the development of an application. 
 
 Researched answer:
+Throughout our challenges over this past week of the cohort, we were using Ruby on Rails packages as we were creating models, working with schemas, and running migrations in our database manipulations. This particular gem provides the essential built-in code needed to connect to and work with backend portion of the application.
```
Comment (SunkissedQueen) [inline interview-questions.md:25]: more content can be to state a ruby gem that you use and talk about what it added to the application

File: interview-questions.md [hunk 5]
```diff
@@ -9,43 +9,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 3. What is Ruby on Rails?
 
 Your answer:
+Rails is the server-based framework that was written in Ruby and is used for backend web development and programming.
 
 Researched answer:
+Ruby on Rails is an MVC (Model, View, Controller) based framework and it gives a developer a different perspective of an application. Development in Rails is occurring on the backend side and it's data as opposed to the aesthetics of design that the front end of the application consists of. Rails is that glue that sticks the front end of an application to the back end of the application and makes for one complete package.
```
Comment (SunkissedQueen) [inline interview-questions.md:33]: more content can be to share how M, V, an C delivers that experience

File: interview-questions.md [hunk 6]
```diff
@@ -9,43 +9,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 4. What is a relational database? Are there other kinds of databases?
 
 Your answer:
+A relational database is the type of database that stores data in tables and gives access to that data through the use of related data points that correspond to the stored data.
 
 Researched answer:
+An active example of a relational database in use would be Amazon. The logisitics of Amazon's business thrives on using relational databases to complete orders by simply matching customer's information (customer_id, mailing address, email) to the actual order information (customer_id, item, cost, payment method) in order to complete a transaction. The customer's info and the order info could stored in different places, yet be connected to each other by a related key reference (customer_id) which brings the information all together to execute a task - sales and product distribution. There are 4 other kinds of databases besides the Relational database: Object-oriented, Network, Hierarchical, and NoSQL.
```
Comment (SunkissedQueen) [inline interview-questions.md:41]: good content here. More things you can share are brief definitions of the 4 other databases you mentioned.

File: interview-questions.md [hunk 7]
```diff
@@ -9,43 +9,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 5. What are primary keys? Why are they important?
 
 Your answer:
+Used in a relational database, primary keys are unique data identifiers and they help to efficiently store, organize, and retrieve data within the data table.
+
 
 Researched answer:
+The primary keyword, better known as the primary key, is important because it is a reference point in the location of data and it is also something that usually never changes. A very common example of a primary key would be the area code of a city. That city's area code, (619) for example would be used as a primary key to reference the data of people that reside within the area that the 619 covers or represents. 
+
```
Comment (SunkissedQueen) [inline interview-questions.md:51]: I understand what is being said. Please consider that rails automatically gives the primary key (PK). So make sure that you are mentioning that when discussing PK. Content that you can add would be a table that shows the primary key as a column.

File: ruby-challenges.rb [hunk 8]
```diff
@@ -9,20 +9,40 @@
...
+    end
+    def get_info
+        "The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph."
+    end
+end
+
+bike1 = Bike.new('Giant')
+p bike1.get_info
 
 # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:47]: Looks like you are missing the pedal_faster and brake methods

**Review by SunkissedQueen — CHANGES_REQUESTED** [review]
Jason, I can complete this assessment after you add the following:

- Test, pseudocode, function for challenge question 3
- pedal_faster and brake method for ruby challenge 3B

**Pull request merged**